calyx-py/calyx/gen_exp.py

```diff
@@ -673,36 +964,145 @@ def generate_exp_taylor_series_approximation(
     parser.add_argument("-w", "--width", type=int)
     parser.add_argument("-i", "--int_width", type=int)
     parser.add_argument("-s", "--is_signed", type=bool)
+    parser.add_argument("-e", "--base_is_e", type=bool)
 
     args = parser.parse_args()
 
-    degree, width, int_width, is_signed = None, None, None, None
-    required_fields = [args.degree, args.width, args.int_width, args.is_signed]
+    degree, width, int_width, is_signed, base_is_e = None, None, None, None, None
+    required_fields = [args.degree, args.width,
+                       args.int_width, args.is_signed, args.base_is_e]
     if all(map(lambda x: x is not None, required_fields)):
         degree = args.degree
         width = args.width
         int_width = args.int_width
         is_signed = args.is_signed
+        base_is_e = args.base_is_e
     elif args.file is not None:
         with open(args.file, "r") as f:
             spec = json.load(f)
             degree = spec["degree"]
             width = spec["width"]
             int_width = spec["int_width"]
             is_signed = spec["is_signed"]
+            base_is_e = spec["base_is_e"]
     else:
         parser.error(
             "Need to pass either `-f FILE` or all of `-d DEGREE -w WIDTH -i INT_WIDTH`"
         )
 
+    # build 2 separate programs: 1 if base_is_e is true, the other if false
+    # any_base_program is (obviously) the one for any base
+    any_base_program = Program(
```

Comment:
Is this all essentially thrown away if the `base_is_e` flag is false? Maybe you want to place this under the if/else as well, to avoid that (perhaps in a new function).

Reply:
Yeah, it' probably cleaner to do it that way. I should update that on this PR before we merge it.